### ISSUE TITLE
@Relation annotation not considered for link relation defaults

### ISSUE THREAD

**Issue by dlynch158** [spring-projects/spring-data-rest#1974]
Hello,

The JSON that gets returned is this:

```
{
    "_embedded": {
        "employees": [
```

I am attempting to change an entity name from employees to "workers" by using the following Relation annotation:

```
@Entity
@Relation(collectionRelation = "workers", itemRelation = "worker")
public class Employee {
```

This has no effect on what gets returned.

The full source code of my test case  is at:

https://github.com/dlynch158/relationtest

Thank you.

**Comment by odrotbohm** [comment]
Looks like this a rather trivial fix as the metadata detection already uses the `LinkRelationProvider` coming from `RepositoryRestConfiguration`, which is currently set up to only use the one based on the Evo Inflector library. Tweaking that to a `DelegatingLinkRelationProvider` with the one considering `@Relation` first is rather easy. That's also a good way of getting the annotation to work with the current state of affairs: just configure an `LRP` as just described via `RepositoryRestConfiguration.setRelProvider(…)`.

**Comment by odrotbohm** [comment]
Fixed for 3.5, 3.4 and 3.3.
